app/Services/BannerFinder.php

```diff
@@ -89,42 +85,57 @@ public static function getBestBanners(array $zones, array $keywords)
+                        $bannerIds[] = '';
+                    }
+                } else {
+                    $bannerIds[] = md5(rand());
                 }
             }
         }
 
         $banners = [];
         foreach ($bannerIds as $bannerId) {
-            // TODO:
-            $banner = $bannerId ? NetworkBanner::where('uuid', hex2bin($bannerId))->first() : null;
-
-            if (!empty($banner)) {
-                $campaign = NetworkCampaign::find($banner->network_campaign_id);
-                $banners[] = [
-                    'serve_url' => $banner->serve_url,
-                    'creative_sha1' => $banner->creative_sha1,
-                    'pay_from' => $campaign->adshares_address, // send this info to log
-                    'click_url' => route('log-network-click', [
-                        'id' => $banner->uuid,
-                        'r' => Utils::urlSafeBase64Encode($banner->click_url),
-                    ]),
-                    'view_url' => route('log-network-view', [
-                        'id' => $banner->uuid,
-                        'r' => Utils::urlSafeBase64Encode($banner->view_url),
-                    ]),
-                ];
-            } else {
-                $banners[] = null;
-                // TODO: discuss Jacek
-            }
+            $banners[] = [
+                'serve_url' => config('app.app_url') . '/img/logo_x.png',
+                'creative_sha1' => sha1_file(public_path('img/logo_x.png')),
+                'pay_from' => AdsUtils::normalizeAddress(config('app.adshares_address')), // send this info to log
+                'click_url' => route('log-network-click', [
+                    'id' => '',
+                    'r' => Utils::urlSafeBase64Encode(config('app.app_url')),
+                ]),
+                'view_url' => route('log-network-view', [
+                    'id' => '',
+                    'r' => Utils::urlSafeBase64Encode(config('app.app_url')),
+                ]),
+            ];
+
+//            // TODO: fix
+//            $banner = $bannerId ? NetworkBanner::where('uuid', hex2bin($bannerId))->first() : null;
+//
+//            if (!empty($banner)) {
+//                $campaign = NetworkCampaign::find($banner->network_campaign_id);
+//                $banners[] = [
+//                    'serve_url' => $banner->serve_url,
+//                    'creative_sha1' => $banner->creative_sha1,
+//                    'pay_from' => $campaign->adshares_address, // send this info to log
+//                    'click_url' => route('log-network-click', [
+//                        'id' => $banner->uuid,
+//                        'r' => Utils::urlSafeBase64Encode($banner->click_url),
+//                    ]),
+//                    'view_url' => route('log-network-view', [
+//                        'id' => $banner->uuid,
+//                        'r' => Utils::urlSafeBase64Encode($banner->view_url),
+//                    ]),
+//                ];
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS125)